crates/evm/src/execute.rs

```diff
@@ -14,33 +19,123 @@ pub use reth_storage_errors::provider::ProviderError;
 ///
 /// The trait cannot commit the state changes to the database directly, see [`EvmCommit`].
 pub trait EvmTransact {
-    /// The environment for the evm.
-    type Env;
-    /// The output produced by the evm.
-    type Output;
-    /// The error produced by the evm.
-    type Error;
+    /// The database used by the evm.
+    type DB: Database;
 
     /// Transacts with the current env to produce an output without
     /// committing to the underlying database.
-    fn transact(&mut self) -> Result<Self::Output, Self::Error>;
+    fn transact(&mut self) -> EVMResult<<<Self as EvmTransact>::DB as Database>::Error>;
```

Comment:
I'm not sure we want to tie the error here to always be DB::Error

Reply:
Is it an issue to have:
```
    /// The output produced by the evm.
    type Output;
    /// The error produced by the evm.
    type Error;
```
?

Reply:
I think that for a first iteration it could be a problem yes because all the returns in reth are configured like that and therefore it would change a lot of things. I figured this was something we could make more generic in the future on other iterations.